packages/cli/src/cmds/beacon/handler.ts

```diff
@@ -28,6 +28,13 @@ export async function beaconHandler(args: IBeaconArgs & IGlobalArgs): Promise<vo
   mkdir(beaconPaths.beaconDir);
   mkdir(beaconPaths.dbDir);
 
+  if (
+    args.dev &&
+    (process.env.NODE_ENV === "dev" || process.env.NODE_ENV === "development" || process.env.NODE_ENV === "test")
+  ) {
+    options = {...options, executionEngine: {mode: "mock"}} as IBeaconNodeOptions;
```

Comment:
This should be a regular CLI option, hidden if you need. Would prefer to not have this ENV switch is a production code path. Just add a hidden option --execution.engineMock